src/Components/CategoryFacet.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import { headlessEngine } from "../Engine";
+import {
+  CategoryFacet as CategoryFacetType,
+  CategoryFacetState,
+  buildCategoryFacet,
+  CategoryFacetValue
+} from "@coveo/headless";
+
+// import { makeStyles } from '@material-ui/core/styles';
+import Typography from "@material-ui/core/Typography";
+import IconButton from "@material-ui/core/IconButton";
+import FormLabel from "@material-ui/core/FormLabel";
+import Box from "@material-ui/core/Box";
+import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+const hoveredStyle = {
+  cursor: "pointer",
+  color: "primary"
+};
+
+interface ICategoryFacetProps {
+  title: string;
+  field: string;
+  subtitle?: string;
+}
+
+export default class CategoryFacet extends React.Component<
+  ICategoryFacetProps,
+  {}
+> {
+  public static defaultProps = {
+    subtitle: "All Categories"
+  };
+
+  private headlessCategoryFacet: CategoryFacetType;
+  state: CategoryFacetState & {
+    inputValue: "";
+  };
+
+  constructor(props: any) {
+    super(props);
+
+    this.headlessCategoryFacet = buildCategoryFacet(headlessEngine, {
+      options: {
+        numberOfValues: 3,
+        field: this.props.field,
+        delimitingCharacter: "|"
```

Comment:
Maybe we could add a comment here to give an example of what a typical `atlgeographicalhierarchy` field value looks like

For instance, the Washington, D.C. result would have `North America;North America|United States;` as an `atlgeographicalhierarchy` field value

Reply:
I've added:
// a typical atlgeographicalhierarchy field value for a city would appear: Continent;Continent|Country;
    // for example Washington, DC. would be: North America;North America|United States;